main.go

```diff
@@ -155,21 +155,26 @@ func main() {
 	Listen(port)
 	fmt.Println("Listening on port " + port + " - http://localhost:" + port + "/\n")
 
-	// TODO: Workaround: this initial call is needed only to populate RateLmitSpecs
-	buildGraph(-1)
+	GetRateLimits(token)
 
 	var duration time.Duration
 	if delay != (60 * pages) {
 		duration = time.Second * time.Duration(delay)
 	} else {
+		if reflect.TypeOf(RateLimitSpecs.PollInterval).String() != "int" || RateLimitSpecs.PollInterval <= 0 {
+			RateLimitSpecs.PollInterval = 60 * pages
+		}
 		duration = time.Second * time.Duration(RateLimitSpecs.PollInterval)
 	}
 
-	// TODO: refreshInterval is constant throughout the whole program. Refactor perhaps?
 	refreshInterval := int64(duration.Seconds())
 
-	lastUpdated, _ := buildGraph(refreshInterval)
+	buildGraph()
+	GetRateLimits(token)
```

Comment:
`GetRateLimits` could be overloaded with an [`http.Headers`](https://godoc.org/net/http#Header) parameter: if given, the call to the RL API endpoint could be skipped and values could be extracted from the given headers, so that no other network calls are made.
Headers might be returned from `buildGraph` or, even better, they could be passed from inside the `unauthenticatedGet` body.